java.lang.VerifyError thrown on nl.b3p.viewer.config.security.User when starting application

### Version
Which version of flamingo do you use: 4.7.10-SNAPSHOT running with tomcat 8.0.38 and Java 8

**Note:** This does not occur on java 7, probably caused by having a default method `public default boolean implies(Subject subject) {...}` in the java.security.Principal interface default methods are only available since Java 8.


### Where?
viewer-admin

### What happened
<!-- Something odd/unexpected/terrible happened. Tell us everything! -->
the following error message is logged:

### What did you expect?
<!-- What should have happened? -->
none of these errors/warnings in the log
 

### Server log
relevant section of the server log:
```
2017-01-09 14:28:40 ERROR org.hibernate.proxy.pojo.BasicLazyInitializer:org.hibernate.proxy.pojo.javassist.JavassistLazyInitializer.getProxyFactory(JavassistLazyInitializer.java:165) - Javassist Enhancement failed: nl.b3p.viewer.config.security.User
java.lang.VerifyError: (class: nl/b3p/viewer/config/security/User_$$_javassist_19, method: _d14implies signature: (Ljavax/security/auth/Subject;)Z) Illegal use of nonvirtual function call
	at java.lang.Class.getDeclaredFields0(Native Method)
	at java.lang.Class.privateGetDeclaredFields(Class.java:2583)
	at java.lang.Class.getField0(Class.java:2975)
	at java.lang.Class.getField(Class.java:1701)
	at javassist.util.proxy.ProxyFactory.setField(ProxyFactory.java:517)
	at javassist.util.proxy.ProxyFactory.createClass3(ProxyFactory.java:502)
	at javassist.util.proxy.ProxyFactory.createClass2(ProxyFactory.java:486)
	at javassist.util.proxy.ProxyFactory.createClass1(ProxyFactory.java:422)
	at javassist.util.proxy.ProxyFactory.createClass(ProxyFactory.java:394)
	at org.hibernate.proxy.pojo.javassist.JavassistLazyInitializer.getProxyFactory(JavassistLazyInitializer.java:162)
	at org.hibernate.proxy.pojo.javassist.JavassistProxyFactory.postInstantiate(JavassistProxyFactory.java:65)
	at org.hibernate.tuple.entity.PojoEntityTuplizer.buildProxyFactory(PojoEntityTuplizer.java:185)
	at org.hibernate.tuple.entity.AbstractEntityTuplizer.<init>(AbstractEntityTuplizer.java:167)
	at org.hibernate.tuple.entity.PojoEntityTuplizer.<init>(PojoEntityTuplizer.java:77)
	at sun.reflect.GeneratedConstructorAccessor58.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.hibernate.tuple.entity.EntityTuplizerFactory.constructTuplizer(EntityTuplizerFactory.java:105)
	at org.hibernate.tuple.entity.EntityTuplizerFactory.constructDefaultTuplizer(EntityTuplizerFactory.java:133)
	at org.hibernate.tuple.entity.EntityEntityModeToTuplizerMapping.<init>(EntityEntityModeToTuplizerMapping.java:80)
	at org.hibernate.tuple.entity.EntityMetamodel.<init>(EntityMetamodel.java:322)
	at org.hibernate.persister.entity.AbstractEntityPersister.<init>(AbstractEntityPersister.java:485)
	at org.hibernate.persister.entity.SingleTableEntityPersister.<init>(SingleTableEntityPersister.java:133)
	at org.hibernate.persister.PersisterFactory.createClassPersister(PersisterFactory.java:84)
	at org.hibernate.impl.SessionFactoryImpl.<init>(SessionFactoryImpl.java:286)
	at org.hibernate.cfg.Configuration.buildSessionFactory(Configuration.java:1872)
	at org.hibernate.ejb.Ejb3Configuration.buildEntityManagerFactory(Ejb3Configuration.java:906)
	at org.hibernate.ejb.HibernatePersistence.createEntityManagerFactory(HibernatePersistence.java:57)
	at javax.persistence.Persistence.createEntityManagerFactory(Persistence.java:63)
	at javax.persistence.Persistence.createEntityManagerFactory(Persistence.java:47)
	at org.stripesstuff.stripersist.Stripersist.init(Stripersist.java:264)
	at org.stripesstuff.stripersist.Stripersist.init(Stripersist.java:157)
	at net.sourceforge.stripes.config.DefaultConfiguration.addInterceptor(DefaultConfiguration.java:487)
	at net.sourceforge.stripes.config.RuntimeConfiguration.initInterceptors(RuntimeConfiguration.java:233)
	at net.sourceforge.stripes.config.RuntimeConfiguration.initInterceptors(RuntimeConfiguration.java:214)
	at net.sourceforge.stripes.config.DefaultConfiguration.init(DefaultConfiguration.java:220)
	at net.sourceforge.stripes.config.RuntimeConfiguration.init(RuntimeConfiguration.java:291)
	at net.sourceforge.stripes.controller.StripesFilter.createConfiguration(StripesFilter.java:137)
	at net.sourceforge.stripes.controller.StripesFilter.init(StripesFilter.java:103)
	at org.apache.catalina.core.ApplicationFilterConfig.initFilter(ApplicationFilterConfig.java:279)
	at org.apache.catalina.core.ApplicationFilterConfig.getFilter(ApplicationFilterConfig.java:260)
	at org.apache.catalina.core.ApplicationFilterConfig.<init>(ApplicationFilterConfig.java:105)
	at org.apache.catalina.core.StandardContext.filterStart(StandardContext.java:4700)
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5340)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145)
	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:725)
	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:701)
	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:717)
	at org.apache.catalina.startup.HostConfig.deployDescriptor(HostConfig.java:587)
	at org.apache.catalina.startup.HostConfig$DeployDescriptor.run(HostConfig.java:1798)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
2017-01-09 14:28:40 WARN  org.hibernate.tuple.entity.PojoEntityTuplizer:org.hibernate.tuple.entity.PojoEntityTuplizer.buildProxyFactory(PojoEntityTuplizer.java:197) - could not create proxy factory for:nl.b3p.viewer.config.security.User
org.hibernate.HibernateException: Javassist Enhancement failed: nl.b3p.viewer.config.security.User
	at org.hibernate.proxy.pojo.javassist.JavassistLazyInitializer.getProxyFactory(JavassistLazyInitializer.java:169)
	at org.hibernate.proxy.pojo.javassist.JavassistProxyFactory.postInstantiate(JavassistProxyFactory.java:65)
	at org.hibernate.tuple.entity.PojoEntityTuplizer.buildProxyFactory(PojoEntityTuplizer.java:185)
	at org.hibernate.tuple.entity.AbstractEntityTuplizer.<init>(AbstractEntityTuplizer.java:167)
	at org.hibernate.tuple.entity.PojoEntityTuplizer.<init>(PojoEntityTuplizer.java:77)
	at sun.reflect.GeneratedConstructorAccessor58.newInstance(Unknown Source)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.hibernate.tuple.entity.EntityTuplizerFactory.constructTuplizer(EntityTuplizerFactory.java:105)
	at org.hibernate.tuple.entity.EntityTuplizerFactory.constructDefaultTuplizer(EntityTuplizerFactory.java:133)
	at org.hibernate.tuple.entity.EntityEntityModeToTuplizerMapping.<init>(EntityEntityModeToTuplizerMapping.java:80)
	at org.hibernate.tuple.entity.EntityMetamodel.<init>(EntityMetamodel.java:322)
	at org.hibernate.persister.entity.AbstractEntityPersister.<init>(AbstractEntityPersister.java:485)
	at org.hibernate.persister.entity.SingleTableEntityPersister.<init>(SingleTableEntityPersister.java:133)
	at org.hibernate.persister.PersisterFactory.createClassPersister(PersisterFactory.java:84)
	at org.hibernate.impl.SessionFactoryImpl.<init>(SessionFactoryImpl.java:286)
	at org.hibernate.cfg.Configuration.buildSessionFactory(Configuration.java:1872)
	at org.hibernate.ejb.Ejb3Configuration.buildEntityManagerFactory(Ejb3Configuration.java:906)
	at org.hibernate.ejb.HibernatePersistence.createEntityManagerFactory(HibernatePersistence.java:57)
	at javax.persistence.Persistence.createEntityManagerFactory(Persistence.java:63)
	at javax.persistence.Persistence.createEntityManagerFactory(Persistence.java:47)
	at org.stripesstuff.stripersist.Stripersist.init(Stripersist.java:264)
	at org.stripesstuff.stripersist.Stripersist.init(Stripersist.java:157)
	at net.sourceforge.stripes.config.DefaultConfiguration.addInterceptor(DefaultConfiguration.java:487)
	at net.sourceforge.stripes.config.RuntimeConfiguration.initInterceptors(RuntimeConfiguration.java:233)
	at net.sourceforge.stripes.config.RuntimeConfiguration.initInterceptors(RuntimeConfiguration.java:214)
	at net.sourceforge.stripes.config.DefaultConfiguration.init(DefaultConfiguration.java:220)
	at net.sourceforge.stripes.config.RuntimeConfiguration.init(RuntimeConfiguration.java:291)
	at net.sourceforge.stripes.controller.StripesFilter.createConfiguration(StripesFilter.java:137)
	at net.sourceforge.stripes.controller.StripesFilter.init(StripesFilter.java:103)
	at org.apache.catalina.core.ApplicationFilterConfig.initFilter(ApplicationFilterConfig.java:279)
	at org.apache.catalina.core.ApplicationFilterConfig.getFilter(ApplicationFilterConfig.java:260)
	at org.apache.catalina.core.ApplicationFilterConfig.<init>(ApplicationFilterConfig.java:105)
	at org.apache.catalina.core.StandardContext.filterStart(StandardContext.java:4700)
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5340)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145)
	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:725)
	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:701)
	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:717)
	at org.apache.catalina.startup.HostConfig.deployDescriptor(HostConfig.java:587)
	at org.apache.catalina.startup.HostConfig$DeployDescriptor.run(HostConfig.java:1798)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.VerifyError: (class: nl/b3p/viewer/config/security/User_$$_javassist_19, method: _d14implies signature: (Ljavax/security/auth/Subject;)Z) Illegal use of nonvirtual function call
	at java.lang.Class.getDeclaredFields0(Native Method)
	at java.lang.Class.privateGetDeclaredFields(Class.java:2583)
	at java.lang.Class.getField0(Class.java:2975)
	at java.lang.Class.getField(Class.java:1701)
	at javassist.util.proxy.ProxyFactory.setField(ProxyFactory.java:517)
	at javassist.util.proxy.ProxyFactory.createClass3(ProxyFactory.java:502)
	at javassist.util.proxy.ProxyFactory.createClass2(ProxyFactory.java:486)
	at javassist.util.proxy.ProxyFactory.createClass1(ProxyFactory.java:422)
	at javassist.util.proxy.ProxyFactory.createClass(ProxyFactory.java:394)
	at org.hibernate.proxy.pojo.javassist.JavassistLazyInitializer.getProxyFactory(JavassistLazyInitializer.java:162)
	... 45 more

```

--- a/pom.xml
+++ b/pom.xml
@@ -275,6 +275,12 @@
                 <version>2.2.3</version>
             </dependency>
             <dependency>
+                <!-- override the version provided by hibernate -->
+                <groupId>org.javassist</groupId>
+                <artifactId>javassist</artifactId>
+                <version>3.22.0-CR1</version>
+            </dependency>
+            <dependency>
                 <groupId>org.hibernate</groupId>
                 <artifactId>hibernate-core</artifactId>
                 <version>3.6.8.Final</version>
@@ -283,6 +289,16 @@
                 <groupId>org.hibernate</groupId>
                 <artifactId>hibernate-entitymanager</artifactId>
                 <version>3.6.8.Final</version>
+                <exclusions>
+                    <exclusion>
+                        <!--
+                        exclude old javassist so we can use 3.22.0-CR1, see issue #745
+                        when upgrading hibernate this should be removed
+                        -->
+                        <groupId>javassist</groupId>
+                        <artifactId>javassist</artifactId>
+                    </exclusion>
+                </exclusions>
             </dependency>
 
             <!-- viewer-config-persistence -->

--- a/solr-commons/pom.xml
+++ b/solr-commons/pom.xml
@@ -36,12 +36,8 @@
         <dependency>
             <groupId>org.stripesstuff</groupId>
             <artifactId>stripersist</artifactId>
-        </dependency>   
-        <dependency>
-            <groupId>org.hibernate</groupId>
-            <artifactId>hibernate-entitymanager</artifactId>
         </dependency>
-        <dependency>
+         <dependency>
             <groupId>org.flamingo-mc</groupId>
             <artifactId>viewer-config-persistence</artifactId>
         </dependency>

--- a/viewer-commons/pom.xml
+++ b/viewer-commons/pom.xml
@@ -76,7 +76,11 @@
         <dependency>
             <groupId>org.stripesstuff</groupId>
             <artifactId>stripersist</artifactId>
-        </dependency>      
+        </dependency>
+        <dependency>
+            <groupId>org.javassist</groupId>
+            <artifactId>javassist</artifactId>
+        </dependency>
         <dependency>
             <groupId>org.hibernate</groupId>
             <artifactId>hibernate-core</artifactId>

--- a/viewer-config-persistence/pom.xml
+++ b/viewer-config-persistence/pom.xml
@@ -146,6 +146,10 @@
             </exclusions>
         </dependency>
         <dependency>
+            <groupId>org.javassist</groupId>
+            <artifactId>javassist</artifactId>
+        </dependency>
+        <dependency>
             <groupId>org.hibernate</groupId>
             <artifactId>hibernate-core</artifactId>
         </dependency>

--- a/viewer-config-persistence/src/main/java/nl/b3p/viewer/config/security/User.java
+++ b/viewer-config-persistence/src/main/java/nl/b3p/viewer/config/security/User.java
@@ -22,7 +22,6 @@ import java.security.NoSuchAlgorithmException;
 import java.security.Principal;
 import javax.persistence.*;
 import java.util.*;
-import javax.security.auth.Subject;
 
 /**
  *
@@ -126,9 +125,4 @@ public class User implements Principal{
     public String getName() {
         return username;
     }
-/*
-    @Override
-    public boolean implies(Subject subject) {
-        return Principal.super.implies(subject); //To change body of generated methods, choose Tools | Templates.
-    }*/
 }

